### PR TITLE
Setup chakra and create game form

### PR DESCRIPTION
**Pull request by HALLERPierre**
## Todo

- [x] Setup chakra ui
- [x] Homepage
- [x] Create party form

image

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,198 +1,75 @@
...
+                            </Flex>
+                        </FormControl>
+                    </form>
+                </Flex>
+            </Flex>
+        </ThemeProvider>
+    );
+};
+
+export default withTheme(Home);
```
Comment (zyhou) [inline pages/index.js:75]: Why are you using `withTheme` here?

File: pages/index.js [hunk 2]
```diff
@@ -1,198 +1,75 @@
...
+        if (!username) {
+            setFormErrors({ username: 'You have to enter an username to play' });
+        } else {
+            const { username: usernameError, ...errors } = formErrors;
+            setFormErrors(errors);
+        }
+    };
+    const handleOnChangeFactory = (field) => (e) => setFormData({ ...formData, [field]: e.target.value });
+    return (
+        <ThemeProvider>
```
Comment (zyhou) [inline pages/index.js:33]: No need you already declare in `_app.js` file

File: pages/index.js [hunk 3]
```diff
@@ -1,198 +1,75 @@
...
+            <Head>
+                <title>Oeaki</title>
+                <link rel="icon" href="/favicon.ico" />
```
Comment (zyhou) [inline pages/index.js:36]: I think we can use the same favicon on all pages

File: pages/index.js [hunk 4]
```diff
@@ -1,198 +1,75 @@
...
+                    </Text>
+                </Heading>
+                <Flex direction="column" justify="center" align="center">
+                    <Text color="gray.700" fontSize="3xl">
+                        Create a game
+                    </Text>
+                    <form onSubmit={handleSubmit}>
+                        <FormControl isInvalid={Object.keys(formErrors).length > 0}>
+                            <Flex direction="column" justify="center" align="center">
+                                <FormLabel htmlFor="username">Username</FormLabel>
```
Comment (zyhou) [inline pages/index.js:53]: `Username` is associate with account. Maybe juste name or pseudo

Comment (zyhou) [reply pages/index.js:53]: Can fix in other PR

File: pages/index.js [hunk 5]
```diff
@@ -1,198 +1,75 @@
...
+        e.preventDefault();
+        const { username } = formData;
+        if (!username) {
+            setFormErrors({ username: 'You have to enter an username to play' });
+        } else {
+            const { username: usernameError, ...errors } = formErrors;
+            setFormErrors(errors);
+        }
+    };
+    const handleOnChangeFactory = (field) => (e) => setFormData({ ...formData, [field]: e.target.value });
```
Comment (zyhou) [inline pages/index.js:31]: No need to add factory, you can keep the name of the field.

```
const handleOnChange= ({ target: { name, value } }) => {
    setFormData({ ...formData, [name]: value });
};
```

File: pages/index.js [hunk 6]
```diff
@@ -1,198 +1,75 @@
...
+    Input,
+} from '@chakra-ui/core';
+import { withTheme } from 'emotion-theming';
+
+const Home = () => {
+    const [formData, setFormData] = useState({
+        username: '',
+    });
+    const [formErrors, setFormErrors] = useState({});
+    const handleSubmit = (e) => {
```
Comment (zyhou) [inline pages/index.js:21]: Can do better, at the moment work welll 👍

File: pages/index.js (hunk 5 above)
Comment (Euregan) [reply pages/index.js:31]: Your version looses the original arity though

**Pull request merged**